packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -451,36 +206,158 @@ export const Tabs = (props: TabsProps) => {
+    onClick: onClickProp,
+    ...rest
+  } = props;
+
+  const {
+    orientation,
+    activeValue: activeValueState,
+    setActiveValue: setActiveValueState,
+    type,
+    setStripeProperties,
+  } = useSafeContext(TabsContext);
+  const { focusActivationMode, tabsWidth } = useSafeContext(TabListContext);
+  const tabRef = React.useRef<HTMLButtonElement>();
+
+  const isActive =
+    isActiveProp !== undefined ? isActiveProp : activeValueState === value;
+
+  const onActiveChange = React.useCallback(() => {
+    setActiveValueState(value);
+    onActiveChangeProp?.();
+  }, [setActiveValueState, value, onActiveChangeProp]);
+
+  useIsomorphicLayoutEffect(() => {
+    if (isActive) {
+      tabRef.current?.scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest',
+        inline: 'nearest',
+      });
     }
+  }, [isActive]);
 
-    const isTabDisabled = (index: number) => {
-      const tab = labels[index];
-      return React.isValidElement(tab) && tab.props.disabled;
-    };
+  const updateStripe = () => {
+    if (type !== 'default' && isActive) {
+      const currentTabRect = tabRef.current?.getBoundingClientRect();
+      setStripeProperties({
+        ...(orientation === 'horizontal' && {
+          '--stripe-width': `${currentTabRect?.width}px`,
+          '--stripe-left': `${tabRef.current?.offsetLeft}px`,
+        }),
+        ...(orientation === 'vertical' && {
+          '--stripe-height': `${currentTabRect?.height}px`,
+          '--stripe-top': `${tabRef.current?.offsetTop}px`,
+        }),
+      });
+    }
+  };
+
+  // CSS custom properties to place the active stripe
+  useIsomorphicLayoutEffect(() => {
+    updateStripe();
+  }, [
+    type,
+    orientation,
+    isActive,
+    tabsWidth, // to fix visual artifact on initial render
+  ]);
 
-    let newIndex = focusedIndex ?? currentActiveIndex;
+  const onKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (event.altKey) {
+      return;
+    }
+    const allTabs = Array.from(
+      event.currentTarget.parentElement?.children ?? [],
+    );
 
     /** focus next tab if delta is +1, previous tab if -1 */
     const focusTab = (delta = +1) => {
-      do {
-        newIndex = (newIndex + delta + labels.length) % labels.length;
-      } while (isTabDisabled(newIndex) && newIndex !== focusedIndex);
-      setFocusedIndex(newIndex);
-      focusActivationMode === 'auto' && onTabClick(newIndex);
+      const currentTab = event.currentTarget as Element;
+
+      let nextTab: Element | undefined = currentTab;
+      if (delta === +1) {
+        nextTab = nextTab?.nextElementSibling ?? allTabs?.at(0);
+      } else if (delta === -1) {
+        nextTab = nextTab?.previousElementSibling ?? allTabs?.at(-1);
+      }
+
+      if (nextTab instanceof HTMLElement) {
+        nextTab?.focus();
+
+        if (focusActivationMode === 'auto') {
+          ReactDOM.flushSync(() => {
+            onActiveChange?.(); // TODO: fix this (stale state)
```

Comment:
this can be fixed using `onFocus`.

<details>
<summary>diff</summary>

```diff
 import React from 'react';
-import * as ReactDOM from 'react-dom';
 import cx from 'classnames';
 import {
   useSafeContext,
@@ -350,13 +349,7 @@ const TabHeader = React.forwardRef((props, forwardedRef) => {
       }
 
       if (nextTab instanceof HTMLElement) {
-        nextTab?.focus();
-
-        if (focusActivationMode === 'auto') {
-          ReactDOM.flushSync(() => {
-            onActiveChange?.(); // TODO: fix this (stale state)
-          });
-        }
+        nextTab.focus();
       }
     };
 
@@ -415,6 +408,11 @@ const TabHeader = React.forwardRef((props, forwardedRef) => {
       tabIndex={isActive ? 0 : -1}
       onClick={mergeEventHandlers(onClickProp, () => onActiveChange?.())}
       onKeyDown={onKeyDown}
+      onFocus={() => {
+        if (focusActivationMode === 'auto') {
+          onActiveChange?.();
+        }
+      }}
       aria-selected={isActive}
       aria-controls={value}
       ref={useMergedRefs(tabRef, forwardedRef, setInitialActiveRef)}
```

</details>

Reply:
Using your suggestion